tests/unit/async_/test_bookmark_manager.py

```diff
@@ -42,243 +42,152 @@ def bookmark_manager(*args, **kwargs):
-    all_bookmarks = await bmm.get_all_bookmarks()
+)
+async def test_chains_bookmarks(update_old, update_new) -> None:
+    initial_bookmarks = [
+        "db1:bm1", "db1:bm1", "db3:bm1", "db3:bm2", "db4:bm4"
+    ]
+    bmm = bookmark_manager(initial_bookmarks=initial_bookmarks)
+    await bmm.update_bookmarks(update_old, update_new)
+    received_bookmarks = await bmm.get_bookmarks()
 
-    assert new_bookmarks == {"db3:bm3", "db3:bm5"}
-    assert all_bookmarks == set(
-        itertools.chain.from_iterable(initial_bookmarks.values())
-    ) | {"db3:bm3", "db3:bm5"}
+    expected_bookmarks = (set(initial_bookmarks)
+                          - set(update_old) | set(update_new))
+    assert sorted(list(received_bookmarks)) == sorted(list(expected_bookmarks))
 
 
 @pytest.mark.parametrize("with_initial_bookmarks", (True, False))
 @pytest.mark.parametrize("consumer_async", consumer_async_options)
-@pytest.mark.parametrize("db", ("db1", "db2", "db3"))
 @mark_async_test
 async def test_notify_on_new_bookmarks(
-    with_initial_bookmarks, consumer_async, db, mocker
+    with_initial_bookmarks, consumer_async, mocker
 ) -> None:
-    initial_bookmarks: t.Dict[str, t.List[str]] = {
-        "db1": ["db1:bm1", "db1:bm1", "db1:bm2"],
-        "db2": ["db2:bm1"],
-    }
+    if with_initial_bookmarks:
+        initial_bookmarks = ["db1:bm1", "db1:bm1", "db1:bm2", "db2:bm1"]
+    else:
+        initial_bookmarks = None
     mock_cls = mocker.AsyncMock if consumer_async else mocker.Mock
     consumer = mock_cls()
     bmm = bookmark_manager(
-        initial_bookmarks=(initial_bookmarks
-                           if with_initial_bookmarks else None),
+        initial_bookmarks=initial_bookmarks,
         bookmarks_consumer=consumer
     )
     bookmarks_old = {"db1:bm1", "db3:bm1"}
     bookmarks_new = {"db1:bm4"}
-    await bmm.update_bookmarks(db, bookmarks_old, bookmarks_new)
+
+    await bmm.update_bookmarks(bookmarks_old, bookmarks_new)
 
     if consumer_async:
         consumer.assert_awaited_once()
         args = consumer.await_args.args
     else:
         consumer.assert_called_once()
         args = consumer.call_args.args
-    assert args[0] == db
-    assert isinstance(args[1], Bookmarks)
-    if with_initial_bookmarks:
-        expected_bms = (
-            set(initial_bookmarks.get(db, [])) - bookmarks_old | bookmarks_new
-        )
-    else:
-        expected_bms = bookmarks_new
-    assert args[1].raw_values == expected_bms
+    assert len(args) == 1
+    assert isinstance(args[0], Bookmarks)
+    expected_bookmarks = (
+        set(initial_bookmarks or []) - bookmarks_old | bookmarks_new
+    )
+    assert args[0].raw_values == expected_bookmarks
 
 
 @pytest.mark.parametrize("consumer_async", consumer_async_options)
 @pytest.mark.parametrize("with_initial_bookmarks", (True, False))
-@pytest.mark.parametrize("db", ("db1", "db2"))
 @mark_async_test
 async def test_does_not_notify_on_empty_new_bookmark_set(
```

Comment:
You won't need to ensure this since the session will never call `update_bookmarks` with `new_bookmarks=[]`.

Reply:
While it's true that the session wouldn't call `update_bookmarks` with empty bookmarks, nothing is stopping user code to do so. `update_bookmarks` is publicly available and documented. No reason not to call it.